examples/db_remember.rs

```diff
@@ -1,24 +1,9 @@
-use teloxide::{
-    dispatching2::dialogue::{serializer::Json, SqliteStorage, Storage},
-    macros::DialogueState,
-    prelude2::*,
-    types::Me,
-    utils::command::BotCommand,
-    RequestError,
-};
-use thiserror::Error;
+use teloxide::{macros::DialogueState, prelude2::*, types::Me, utils::command::BotCommand};
 
-type MyDialogue = Dialogue<State, SqliteStorage<Json>>;
-type StorageError = <SqliteStorage<Json> as Storage<State>>::Error;
+// include!("redis_config.in");
+include!("sqlite_config.in");
```

Comment:
I don't think `include!` is the right thing here it's confusing and rarely used in Rust. Using `Arc<dyn Storage>` and then choosing DB at runtime would be better IMO.

Reply:
If the problem is the `Error` type, we can add some erasing like in `teloxide-core` for `Requester`.

Maybe something like this in the `Storage` trait:
```rust
fn erase(self: Arc<Self>) -> Arc<dyn Storage<Error = Box<dyn std::error::Error>> {
    struct Eraser<T>(Arc<T>);

    impl<T: Storage> Storage for Eraser { ... }

    Arc::new(Eraser(self))
}
```

Or maybe return something like `ErasedStorage` that holds `Arc<dyn Storage<Error = Box<dyn std::error::Error>>` inside, idk.

Reply:
FIxed.